lib/modules/manager/gomod/artifacts.ts

```diff
@@ -177,7 +177,10 @@ export async function updateArtifacts({
     // replace (
     //     golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5
     // )
-    const blockReplaceRegEx = regEx(/(\r?\n)replace\s*\([^)]+\s*\)/g);
+    // the regex will ignore commented lines with "//"
+    const blockReplaceRegEx = regEx(
+      /replace\s*\((?:(?:\s*\/\/.*)?(?:[^)])?)+\)/g
```

Comment:
```suggestion
      /\r?\nreplace\s*\((?:\r?\n\s+\/\/[^\n]+)*[^)]+\)/g
```

what about this?

- https://regex101.com/r/WN0M0H/1

Reply:
it fails on double comments in different lines
```
replace (
  go.uber.org/zap => go.uber.org/zap v1.21.0
  // TODO: this comment breaks renovatebot (>v0.11.1)
  go.uber.org/zap => go.uber.org/zap v1.21.0
  // TODO: this comment breaks renovatebot (>v0.11.1)
  go.uber.org/zap => go.uber.org/zap v1.21.0
)
```

Reply:
do you think i should just look for all comments in file and replace them with Spaces for the sake of replace regex? then get it back? instead of doing such complex regex?

Reply:
no.

@zharinov It seems `moo` lexer is a better option here? As the regex will be too complex

Reply:
Looks like it is

Reply:
Can you do some suggestions?

Reply:
I think, the best example would be `lib/modules/manager/cake/index.ts`

Reply:
do you mean this? let url = line.replace(regEx(/^[^:]*:/), '');

where you replace comments with nothing?

Reply:
you mean when using lexer?
```
export function extractPackageFile(content: string): PackageFile {
  const deps: PackageDependency[] = [];
  lexer.reset(content);
  let token = lexer.next();
  while (token) {
    const { type, value } = token;
    if (type === 'dependency' || type === 'dependencyQuoted') {
      const dep = parseDependencyLine(value);
      if (dep) {
        deps.push(dep);
      }
    }
    token = lexer.next();
  }
  return { deps };
}
```

Reply:
Can we have a short term solution though? This is not a hugely high priority item for me, so happy with a quick fix

Reply:
Why the comment removal before any processing isn't option? If I understand correctly, we don't care about offsets info here.

Reply:
if we remove anything we will have to replace it back since the changes affect the PR in the end

Reply:
i'm not 100% familiar with go, so are you sure the Command mod tidy doesn't update anything?
What if the command changes the file?
i think we are doing some changes:
if you look here
```
  if (finalGoModContent !== newGoModContent) {
      logger.debug('Found updated go.mod after go.sum update');
```
apparently we do some changes.

Unless i am fully familiar with the flow i cant say.